baybe/recommenders/base.py

```diff
@@ -138,28 +147,29 @@ def recommend(  # noqa: D102
                 searchspace,
                 self._recommend_discrete,
                 batch_quantity,
-                allow_repeated_recommendations,
-                allow_recommending_already_measured,
+                self.allow_repeated_recommendations,
+                self.allow_recommending_already_measured,
             )
         if searchspace.type == SearchSpaceType.CONTINUOUS:
             return self._recommend_continuous(
-                searchspace=searchspace, batch_quantity=batch_quantity
+                subspace_continuous=searchspace.continuous,
```

Comment:
Is there any specific reason why these changes are made in this PR? To me, this feels like something rather unrelated which should thus be done in a different PR.

Reply:
please see see Azure work item 15200
these are also recommender refactoring so why do yous ay they are unrelated?

Reply:
Well, this is a *different* change and refactoring and not related to unifying strategies and recommenders

Reply:
ok ive mentioned that in the PR description the 2 other refactoings were put into 1 commit each, if you insist they could be cherrypicked and put into another PR

Reply:
I'd personally prefer to have this as a separate PR, but I am also fine with keeping it here. What is @AdrianSosic opinion here?

Reply:
In the general case, I'd be with @AVHopp here, but since it's already done and I've also put more changes on top during my review, I think it would now be more dangerous to revert than just keeping the commits in this PR. So let's be pragmatic here, save some effort, and eat cookies instead 🍪

Reply:
then lets just finish this PR ill resolve